test/unit/descriptors.spec.ts

```diff
@@ -0,0 +1,92 @@
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", { writable: ${value} });
+            try { foo.bar = false } catch {}
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test.each(trueFalseUndefTests)("configurable (%p)", value => {
+        util.testFunction`
+            const foo = { bar: true };
+            Object.defineProperty(foo, "bar", { configurable: ${value} });
+            try { delete foo.bar } catch {}
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("defines a new property", () => {
+        util.testFunction`
+            const foo: any = {};
+            Object.defineProperty(foo, "bar", { value: true });
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("overwrites an existing property", () => {
+        util.testFunction`
+            const foo = { bar: false };
+            Object.defineProperty(foo, "bar", { value: true });
+            return foo.bar;
+        `.expectToMatchJsResult();
+    });
+
+    test("default descriptor", () => {
+        util.testFunction`
+            const foo = {};
+            Object.defineProperty(foo, "bar", {});
+            return Object.getOwnPropertyDescriptor(foo, "bar");
+        `.expectToMatchJsResult();
+    });
+
+    test.each([
+        ["{ value: true, get: () => true }"],
+        ["{ value: true, set: value => {} }"],
+        ["{ writable: true, get: () => true }"],
+    ])("invalid descriptor (%p)", props => {
+        util.testFunction`
+            const foo: any = {};
+            let err = false;
+
+            try {
+                Object.defineProperty(foo, "bar", ${props});
+            } catch {
+                err = true;
+            }
+
+            return { prop: foo.bar, err };
+        `.expectToMatchJsResult();
+    });
+});
+
+describe("Decorators /w descriptors", () => {
+    test.each([
+        ["return { writable: true }", "return { configurable: true }"],
+        ["desc.writable = true", "return { configurable: true }"],
+    ])("Combine decorators (%p + %p)", (decorateA, decorateB) => {
+        util.testFunction`
+            const A = (target, key, desc): any => { ${decorateA} };
+            const B = (target, key, desc): any => { ${decorateB} };
+            class Foo { @A @B static method() {} }
+            const { value, ...rest } = Object.getOwnPropertyDescriptor(Foo, "method");
+            return rest;
+        `.expectToMatchJsResult();
+    });
+
+    test.each(["return { value: true }", "desc.value = true"])(
+        "Use decorator to override method value",
+        overrideStatement => {
+            util.testFunction`
+                const decorator = (target, key, desc): any => { ${overrideStatement} };
+                class Foo { @decorator static method() {} }
+                return Foo.method;
+            `.expectToMatchJsResult();
```

Comment:
Maybe it would be nice to also have a `.expectToEqual(true)` for this one

Reply:
I used this one thinking

- More test cases could be added, some which don't just change `Foo.method` to `true`
- To ensure no new transpiler specific behaviour is introduced, match what JS does

I think these justify the `expectToMatchJsResult`, what do you think?